### DISCUSSION TITLE
How different ldap implementations are generating random salt?

### DISCUSSION THREAD

**Discussion by Karan-nayyar** [OpenIdentityPlatform/OpenDJ#210] [Q&A]
I am currently generating a salted SHA 256 passwords in the below format

`$hash = "{SSHA256}".base64_encode(hash('sha256', $password . $salt) . $salt) .
`

Using the below libraries of Java classes to generate.

```
java.security.MessageDigest
java.security.SecureRandom

```
I am not sure if the random salt generated via SecureRandom class is secure enough.I wanted to know how different ldap implementations(OpenLDAP,OpenDJ) are generating the random salt for different password storage scheme like SHA-1,SHA256 etc. Also, can those salt generation algorithm used by different ldap implementations be utilised by us in our scripts(using js) to generate same. I am using js scripts (Rhino js engine) to generate salted sha256 password. Are there any other random generators which are secure enough and can be used in our application?

**Comment by vharseko** [comment]
java.security.SecureRandom
